nestedtensor/csrc/UnaryOps.cpp

```diff
@@ -0,0 +1,235 @@
+#include <ATen/core/op_registration/op_registration.h>
+#include <nestedtensor/csrc/nested_tensor_impl.h>
+#include <torch/library.h>
+
+namespace at {
+
+using namespace torch::nested_tensor;
+
+// NOTE: Can't reuse dispatch from cos_ to cos_out either, because it requries
+// support for at::empty through unary_op_impl
+template <class F, F func>
+Tensor& NestedTensor_unary_(Tensor& self) {
+  auto self_impl = get_nested_tensor_impl(self);
+  auto f = [](at::Tensor& tensor) { func(tensor); };
+  apply<decltype(f)>(std::move(f), self_impl->_data.get_structure());
+  return self;
+}
+
+template <class F, F func>
+Tensor NestedTensor_unary(const Tensor& self) {
+  auto self_impl = get_nested_tensor_impl(self);
+  return at::detail::make_tensor<NestedTensorImpl>(
+      map([](at::Tensor tensor) { return func(tensor); },
+          self_impl->_data.get_structure()));
+}
+
+template <class F, F func>
+Tensor& NestedTensor_unary_out(Tensor& result, const Tensor& self) {
+  auto result_impl = get_nested_tensor_impl(result);
+  auto self_impl = get_nested_tensor_impl(self);
+  apply([](at::Tensor& result, const at::Tensor tensor)
+          { return func(result, tensor); },
+      result_impl->_data.get_structure(),
+      self_impl->_data.get_structure());
+  return result;
+}
+
+Tensor& NestedTensor_clamp_(Tensor& self, optional<Scalar> min, optional<Scalar> max) {
+  auto self_impl = get_nested_tensor_impl(self);
+  auto f = [min, max](at::Tensor& tensor) { at::clamp_(tensor, min, max); };
+  apply<decltype(f)>(std::move(f), self_impl->_data.get_structure());
+  return self;
+}
+
+Tensor NestedTensor_clamp(const Tensor& self, optional<Scalar> min, optional<Scalar> max) {
```

Comment:
The thing about the `clamp` functions is that `clamp_min` and `clamp_max` are implementation details. I don't think you need them for nested tensor (?) because it looks like the code is just mapping clamp onto the tree structure

Reply:
Although we expose torch.clamp_min and torch.clamp_max, only torch.clamp should really be used :/ (exposing those were an accident imo)

Reply:
This is one of those things we might want to consider prefixing clamp_min / clamp_max it with an underscore etc. But dir(torch) has a lot of stuff that's undocumented, I think simply because that's how we deal with registering native functions (which seems reasonable).